cardinal/server/server.go

```diff
@@ -232,14 +235,45 @@ func (handler *Handler) Serve() error {
 		return eris.Wrap(err, "error getting hostname")
 	}
 	log.Info().Msgf("serving cardinal at %s:%s", hostname, handler.Port)
-	return eris.Wrap(handler.server.ListenAndServe(), "error listening and serving")
+	err = eris.Wrap(handler.server.ListenAndServe(), "error listening and serving")
+	if err != nil {
+		return err
+	}
+	handler.running.Store(true)
```

Comment:
ListenAndServe only returns non-nil errors, and blocks while everything is healthy. The Store(true) here will never be called.

Reply:
I found some indeterministic output from the logging that I only caught after running this several times. I put in some mutexes to serialize shutdown calls. But this part has been changed.